### ISSUE TITLE
Sharp as a dependency of Ghost, install fails on upgrade

### ISSUE THREAD

**Issue by ErisDS** [lovell/sharp#1941]
One of our users is having problems getting sharp reinstalled after upgrading Node.js. Originally reported here: https://forum.ghost.org/t/after-update-to-version-3-ghost-wont-start/9677/5

When running `sudo npm rebuild` inside the ghost folder, they get this error:

```
info sharp Using existing vendored libvips v8.8.1
Traceback (most recent call last):
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py", line 47, in <module>
    import gyp
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 10, in <module>
    import gyp.input
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/input.py", line 7, in <module>
    import ast
ImportError: No module named ast
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:351:16)
gyp ERR! stack     at ChildProcess.emit (events.js:210:5)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Linux 4.15.0-23-generic
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /var/www/ghostname.com/html/blog/versions/3.0.0/node_modules/sharp
gyp ERR! node -v v12.13.0
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! sharp@0.23.1 install: `(node install/libvips && node install/dll-copy && prebuild-install) || (node-gyp rebuild && node install/dll-copy)`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the sharp@0.23.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

I've searched github for this error and couldn't find any related reports. What would cause this to fail on a machine where sharp was previously installed?

**Comment by lovell** [comment]
Hello, if you're using `npm install` or `npm rebuild` with globally-installed native modules via `sudo` then you'll need to pass the `npm --unsafe-perm` flag.

As an aside, I notice Ghost currently recommends the global installation of `ghost-cli`. Has the use of a package manager tool such as  `npx ghost ...` or `yarn dlx ghost ...` been considered? These are designed to improve both the user experience and security of running lifecycle CLI tools.

**Comment by lovell** [comment]
When npm is run as sudo or root with elevated privileges, it drops these privileges before running install scripts within modules.

All native modules use install scripts to either (attempt to) download prebuilt binaries or build from source (via node-gyp and binding.gyp) and therefore need to write files into `node_modules` so still require the elevated privileges. The `--unsafe-perm` flag prevents npm from dropping these privileges.

The example `npx` command I gave had the wrong package name, sorry for the confusion. By way of example, on a machine that does not have `ghost-cli` installed, I can successfully run:

```sh
$ npx ghost-cli version
npx: installed 383 in 11.898s
Ghost-CLI version: 1.12.0
```

This ensures the most recent version of ghost-cli is always run, as the current user, without any sudo/root privileges.

**Comment by lovell** [comment]
I think this can be closed like #1942 as the various logging improvements should help expose the underlying problem(s) if it happens again. As with that issue, please feel free to re-open this issue with more info if required.
